examples/rllib/self_play_train.py

```diff
@@ -60,22 +54,29 @@ def env_creator(env_config):
   # 5. Configuration for multiagent setup with policy sharing:
   config["multiagent"] = {
       "policies": {
-          # the first tuple value is None -> uses default policy
-          "av": (None, obs_space, act_space, {}),
+          "av":
+              PolicySpec(
+                  policy_class=None,  # use default policy
+                  observation_space=obs_space,
+                  action_space=act_space,
+                  config={}),
       },
       "policy_mapping_fn": lambda agent_id, **kwargs: "av"
   }
 
   # Use GPUs iff `RLLIB_NUM_GPUS` env var set to > 0.
   config["num_gpus"] = int(os.environ.get("RLLIB_NUM_GPUS", "0"))
   config["log_level"] = "DEBUG"
-  config["num_workers"] = 1
   # Fragment length, collected at once from each worker and for each agent!
-  config["rollout_fragment_length"] = 30
+  config["rollout_fragment_length"] = 100
   # Training batch size -> Fragments are concatenated up to this point.
-  config["train_batch_size"] = 200
+  # Setting this longer than the horizon ensures that we have a full episode
+  # for computing episodic metrics
+  config["train_batch_size"] = 2 * config["env_config"].lab2d_settings[
```

Comment:
Is there an adverse impact of setting this to 2x the max episode length? Could we just set it to the max length itself?

Reply:
Also, this is the train batch size, so probably we don't want it to depend on the episode length, right? This is a different dimension (e.g. batch x time x other dims).

Reply:
I am under the impression that the train batch size is just the number of environment steps collected before a training step, so not the size of the batch going into the GPU. Therefore I just selected a value that would be guaranteed to work. Will confirm this however...

Reply:
Yes, this appears to be the case (see rllib.execution.sgd.train_one_step), but I will need to explicitly set sgd_minibatch_size. By default it is equal to train_batch_size, which is what you were concerned about.

It is important that we have at least a full episode however, otherwise the rewards returned are nan. I will instead change to counting the number of episodes to make things clearer.